lib/src/html_input_stream.dart

```diff
@@ -20,28 +20,28 @@ class HtmlInputStream {
   static const String defaultEncoding = 'utf-8';
 
   /// The name of the character encoding.
-  String charEncodingName;
+  String? charEncodingName;
 
   /// True if we are certain about [charEncodingName], false for tenative.
   bool charEncodingCertain = true;
 
   final bool generateSpans;
 
   /// Location where the contents of the stream were found.
-  final String sourceUrl;
+  final String? sourceUrl;
 
-  List<int> _rawBytes;
+  List<int>? _rawBytes;
 
   /// Raw UTF-16 codes, used if a Dart String is passed in.
-  List<int> _rawChars;
+  List<int>? _rawChars;
 
-  Queue<String> errors;
+  var errors = Queue<String>();
```

Comment:
Public interface declarations should have types on the left. Using `var` instead is only for local variables.

Reply:
`var` is allowed for fields and top-level variables as well when the type on the right is "obvious" which includes constructor calls.

https://dart.dev/guides/language/effective-dart/design#prefer-type-annotating-public-fields-and-top-level-variables-if-the-type-isnt-obvious

We typically don't comment if it's included on the left because some teams prefer that, but in general it seems to me like the trend is away from types on the left.